numba/tests/test_casting.py

```diff
@@ -80,6 +84,19 @@ def driver(x):
         self.assertEqual(driver(x), x[0])
         self.assertEqual(len(inner.overloads), 1)
 
+    def test_array_to_scalar(self):
+        """
+        Ensure that a TypingError exception is raised if
+        user tries to convert numpy array to scalar
+        """
+
+        try:
+            compile_isolated(numpy_scalar_cast_error, ())
+        except TypingError as e:
+            self.assertIn("Casting array", str(e))
+        else:
+            self.fail("Should raise error")
```

Comment:
You can use [assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises) to avoid having to do an explicit `try..except` and `self.fail()`.

Reply:
@sklam thanks for the comment, code has been updated.

Reply:
we should still have the `assertIn` to make sure it is getting the right exception. 

see example: https://github.com/numba/numba/blob/bb2ec93fe107339d752bb03bde49f505fa2ed02c/numba/cuda/tests/cudadrv/test_context_stack.py#L57-L60

Reply:
@sklam your request was incorporated in the most recent commits.